File: Factory Method/Java/src/factory/UserFactory.java
package factory;

import factory.users.Normal;
import factory.users.Premium;

public class UserFactory {
    public static User getUser(UserType userType, String name, String surname) {
        switch (userType) {
            case Normal:
                return new Normal(name, surname);
            case Premium:
                return new Premium(name, surname);
            default:
                return new Normal(name, surname);
        }
    }
}

File: Factory Method/Kotlin/src/factory/UserFactory.kt
package factory

object UserFactory {
    fun getUser(userType: UserType, name: String, surname: String): User {
        return when (userType) {
            UserType.Normal -> Normal(name = name, surname = surname)
            UserType.Premium -> Premium(name = name, surname = surname)
        }
    }
}